autopush_rs/src/settings.rs

```diff
@@ -0,0 +1,130 @@
+use std::net::ToSocketAddrs;
+
+use config::{Config, ConfigError, Environment, File};
+use fernet::Fernet;
+use hostname::get_hostname;
+
+lazy_static! {
+    static ref HOSTNAME: String = get_hostname().unwrap();
+    static ref RESOLVED_HOSTNAME: String = get_resolved_hostname();
+}
+
+fn get_resolved_hostname() -> String {
+    let hostname = get_hostname().expect("Can't get hostname");
+    hostname
+        .to_socket_addrs()
+        .expect("Failed to resolve hostnames")
+        .last()
+        .expect("No hostnames found")
+        .to_string()
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    pub debug: bool,
+    pub port: u16,
+    pub hostname: Option<String>,
+    pub resolve_hostname: bool,
+    pub router_port: u16,
+    pub router_hostname: Option<String>,
+    pub router_tablename: String,
+    pub message_tablename: String,
+    pub router_ssl_key: Option<String>,
+    pub router_ssl_cert: Option<String>,
+    pub router_ssl_dh_param: Option<String>,
+    pub auto_ping_interval: f64,
+    pub auto_ping_timeout: f64,
+    pub max_connections: u32,
+    pub close_handshake_timeout: u32,
+    pub endpoint_scheme: String,
+    pub endpoint_hostname: Option<String>,
+    pub endpoint_port: u16,
+    pub crypto_key: String,
```

Comment:
this is an append (List[str]) in main_argparse. Can Config do something similar?

Reply:
Not if we want it to work with env vars.